open_samus_returns_rando/specific_patches/map_icons.py

```diff
@@ -0,0 +1,332 @@
+    mapicon[100]["icons"][0]["icon"] = "doorpowerleft"
+    mapicon[101]["icons"][0]["icon"] = "doorpowerright"
+
+    # Items
+    for tile in range(296):
+        if mapicon[tile]["icons"] != ([]) and mapicon[tile]["icons"][0][
+            "icon"
+        ].startswith("item"):
+            mapicon[tile]["icons"][0]["icon"] = "itemenabled"
+    mapicon[213]["icons"][0]["icon"] = "itemenabledheat"
+
+
+def _area6_icons(editor: PatcherEditor):
+    area6 = editor.get_file("gui/minimaps/c10_samus/s060_area6.bmsmsd", Bmsmsd)
+    mapicon = area6.raw["tiles"]
+
+    # Doors
+    # Door001 (Entrance Charge Door)
+    mapicon[14]["icons"][0]["icon"] = "doorpowerleft"
+    mapicon[15]["icons"][0]["icon"] = "doorpowerright"
+
+    for tile in range(226):
+        # Hazardous Tiles
+        if mapicon[tile]["tile_type"] == "ACID":
+            mapicon[tile]["tile_type"] = "NORMAL"
+        # Items
+        if mapicon[tile]["icons"] != ([]) and (
+            mapicon[tile]["icons"][0]["icon"].startswith("item")
+            or mapicon[tile]["icons"][0]["icon"].startswith("spenergy")
+        ):
+            mapicon[tile]["icons"][0]["icon"] = "itemenabled"
+
+
+def _area6b_icons(editor: PatcherEditor):
+    area6b = editor.get_file("gui/minimaps/c10_samus/s065_area6b.bmsmsd", Bmsmsd)
+    mapicon = area6b.raw["tiles"]
+
+    # Items
+    for tile in range(290):
+        if mapicon[tile]["icons"] != ([]) and mapicon[tile]["icons"][0][
+            "icon"
+        ].startswith("item"):
+            mapicon[tile]["icons"][0]["icon"] = "itemenabled"
+
+
+def _area6c_icons(editor: PatcherEditor):
+    area6c = editor.get_file("gui/minimaps/c10_samus/s067_area6c.bmsmsd", Bmsmsd)
+    mapicon = area6c.raw["tiles"]
+
+    # Doors
+    # Door005 (Plasma Beam Missile Door)
+    mapicon[32]["icons"][0]["icon"] = "doorpowerleft"
+
+    # Door006 (Plasma Beam Plasma Door)
+    mapicon[35]["icons"][0]["icon"] = "doorpowerleft"
+
+    # Door009 (Gravity Suit Missile Door)
+    mapicon[149]["icons"][0]["icon"] = "doorpowerleft"
+
+    # Items
+    for tile in range(212):
+        if mapicon[tile]["icons"] != ([]) and mapicon[tile]["icons"][0][
+            "icon"
+        ].startswith("item"):
+            mapicon[tile]["icons"][0]["icon"] = "itemenabled"
+
+
+def _area7_icons(editor: PatcherEditor):
+    area7 = editor.get_file("gui/minimaps/c10_samus/s070_area7.bmsmsd", Bmsmsd)
+    mapicon = area7.raw["tiles"]
+
+    # Doors
+    # Door010 (Pre-Steal Charge Door)
+    mapicon[273]["icons"][0]["icon"] = "doorpowerleft"
+    mapicon[274]["icons"][0]["icon"] = "doorpowerright"
+
+    for tile in range(351):
+        # Hazardous Tiles
+        if mapicon[tile]["tile_type"] in {"ACID", "ACID_RISE", "ACID_FALL"}:
+            mapicon[tile]["tile_type"] = "NORMAL"
+        # Items
+        if mapicon[tile]["icons"] != ([]) and mapicon[tile]["icons"][0][
+            "icon"
+        ].startswith("item"):
+            mapicon[tile]["icons"][0]["icon"] = "itemenabled"
```

Comment:
Is it possible to make the code less verbose?
Some check for `ACID_RISE` and `ACID_FALL`, some not, but can you just do it for all scenarios? Is the `range` parameter just the length?
I'm thinking about one function which get's called for every bmsmsd. If I combined it correctly it would look like this?
```python
def bla(tiles: ListContainer):
    for tile_idx in range(len(tiles)):
        current_tile = tiles[tile_idx]
        # Hazardous Tiles
        if current_tile["tile_type"] in {"ACID", "ACID_RISE", "ACID_FALL"}:
            current_tile["tile_type"] = "NORMAL"

        # Items
        icons = current_tile["icons"]
        if len(icons) != 0 and (
            icons[0]["icon"].startswith("item")
            or icons[0]["icon"].startswith("spenergy")
            or icons[0]["icon"].startswith("itemsphere")
            or icons[0]["icon"].endswith("tank")
        ):
            icons[0]["icon"] = "itemenabled"
```

Reply:
Yes the range is the length